modules/ROOT/pages/functions.adoc

```diff
@@ -0,0 +1,143 @@
+= Luna Streaming Functions
+
+Functions are lightweight compute processes that enable you to process each message received on a topic. You can apply custom logic to that message, transforming or enriching it, and then output it to a different topic.
+
+Functions run inside Luna Streaming and are therefore serverless. You write the code for your function in Java or Python, then upload the code. It will be automatically run for each message published to the specified input topic.
+
+Functions are implemented using Apache Pulsar(R) functions. See https://pulsar.apache.org/docs/en/functions-overview/[Pulsar Functions overview] for more information about Pulsar functions.
+
+== Add Functions in Luna Streaming
+
+Add functions in the *Functions* tab of the Admin Console. 
+
+. Select *Choose File* to choose a local Function. In this example, we chose `exclamation_function.py`.
+. Choose your function's name and namespace. 
++
+image::admin-console-add-function.png[Add Function in Admin Console]
+[start=3]
+. Choose the file you want to pull the function from and which function you want to use within that file.
+
+Luna generates a list of acceptable classes. Python and Java functions are added a little differently from each other.
```

Comment:
Change to something like: "To specify the class name, it depends on the programming language (or runtime) of the function.